addons/base_sparse_field/models/models.py

```diff
@@ -29,32 +31,45 @@ def write(self, vals):
 
         return super(IrModelFields, self).write(vals)
 
-    def _reflect_model(self, model):
-        super(IrModelFields, self)._reflect_model(model)
+    def _reflect_fields(self, model_names):
+        super()._reflect_fields(model_names)
 
         # set 'serialization_field_id' on sparse fields; it is done here to
         # ensure that the serialized field is reflected already
         cr = self._cr
-        query = """ UPDATE ir_model_fields
-                    SET serialization_field_id=%s
-                    WHERE model=%s AND name=%s
-                    RETURNING id
-                """
-        fields_data = self._existing_field_data(model._name)
-
-        for field in model._fields.values():
-            ser_field_id = None
-            if field.sparse:
-                if field.sparse not in fields_data:
-                    msg = _("Serialization field `%s` not found for sparse field `%s`!")
-                    raise UserError(msg % (field.sparse, field.name))
-                ser_field_id = fields_data[field.sparse]['id']
-
-            if fields_data[field.name]['serialization_field_id'] != ser_field_id:
-                cr.execute(query, (ser_field_id, model._name, field.name))
-                record = self.browse(cr.fetchone())
-                self.pool.post_init(record.modified, ['serialization_field_id'])
-                self.clear_caches()
+
+        # retrieve existing values
+        query = """
+            SELECT model, name, id, serialization_field_id
+            FROM ir_model_fields
+            WHERE model IN %s
+        """
+        cr.execute(query, [tuple(model_names)])
+        existing = {row[:2]: row[2:] for row in cr.fetchall()}
+
+        # determine updates, grouped by value
+        updates = defaultdict(list)
+        for model_name in model_names:
+            for field_name, field in self.env[model_name]._fields.items():
+                field_id, current_value = existing[(model_name, field_name)]
+                try:
+                    value = existing[(model_name, field.sparse)][0] if field.sparse else None
+                except KeyError:
+                    msg = _("Serialization field %r not found for sparse field %s!")
+                    raise UserError(msg % (field.sparse, field))
+                if current_value != value:
+                    updates[value].append(field_id)
+
+        if not updates:
+            return
+
+        # update fields
+        query = "UPDATE ir_model_fields SET serialization_field_id=%s WHERE id IN %s"
+        for value, ids in updates.items():
+            cr.execute(query, [value, tuple(ids)])
```

Comment:
`execute_batch` might be an option here depending how many updates we usually have. I *think* psycopg 2.7 is required on master / 13.3 so it ought be available.